lib/Dialect/ModArith/IR/ModArithOps.cpp

```diff
@@ -214,17 +214,33 @@ void ConstantOp::print(OpAsmPrinter &p) {
 
 // constant(c0) -> c0 mod q
 OpFoldResult ConstantOp::fold(FoldAdaptor adaptor) {
-  // Check if the value is an IntegerAttr
-  auto intAttr = dyn_cast_if_present<IntegerAttr>(adaptor.getValue());
-  if (!intAttr) return {};
-
-  auto modType = dyn_cast_if_present<ModArithType>(getType());
+  Type type = getElementTypeOrSelf(getType());
+  auto modType = dyn_cast<ModArithType>(type);
+  auto storageType = modType.getModulus().getType();
   if (!modType) return {};
```

Comment:
```suggestion
  if (!modType) return {};
  auto storageType = modType.getModulus().getType();
```
Not sure `modType` can actually ever be null (would make the op invalid w.r.t. to verifier?) but probably safer this way around.

Reply:
Done